confidential-datahub/kms/src/plugins/mod.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::{Decryptor, Error, ProviderSetting, Result};
+
+#[cfg(feature = "aliyun")]
+pub mod aliyun;
+
+/// Create a new [`Decryptor`] by given provider name and [`ProviderSetting`]
+pub async fn new_decryptor(
+    provider: &str,
+    provider_setting: ProviderSetting,
+) -> Result<Box<dyn Decryptor>> {
+    match provider {
+        "ali" => Ok(Box::new(
+            aliyun::AliyunKmsClient::from_provider_setting(&provider_setting).await?,
+        ) as Box<dyn Decryptor>),
```

Comment:
Do we want a feature flag around this condition?

Reply:
Oh, yes! Thanks for pointing out!